deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -134,16 +146,15 @@ public void waitForDeployment(DeploymentContext context) throws HyscaleException
         deltaWait(DELTA_WAIT_MILLIS);
         // Pod Scheduling
         ActivityContext podInitContext = new ActivityContext(DeployerActivity.POD_INITIALIZED);
-        waitForPodActivity(apiClient, podHandler, namespace, selector, PodPredicates.isPodschedulingCondition(),
-                podInitContext);
-
+        try {
+            podHandler.watch(apiClient,appName, serviceName, namespace);
+		} catch (Exception e) {
+			throw new HyscaleException(DeployerErrorCodes.FAILED_TO_INITIALIZE_POD);
+		}
         // Pod Creation
         ActivityContext podCreationContext = new ActivityContext(DeployerActivity.POD_CREATION);
-        waitForPodActivity(apiClient, podHandler, namespace, selector, PodPredicates.isPodCreated(),
-                podCreationContext);
         // Pod readiness
         ActivityContext podReadinessContext = new ActivityContext(DeployerActivity.POD_READINESS);
```

Comment:
These activity contexts are being in use ? 
Ideally they should be  used in watch events